CriteoPublisherSdk/Sources/Standalone/CRBannerView.m

```diff
@@ -165,6 +174,11 @@ - (void)loadAdWithBid:(CRBid *)bid {
     return;
   }
 
+  if (self.adUnit.adUnitId == nil && [bid.adUnit isKindOfClass:[CRBannerAdUnit class]]) {
```

Comment:
On `bid.adUnit isKindOf Banner`:

Ok to have such condition because you need the size info. But I would ask: what should we do if it is not the expected kind ? In your dev, in such unexpected case, we just continue displaying and notifying the publisher that everything was good.

From such condition, I expect a `else` branch with a `self.safelyNotifyAdLoadFail:CRErrorCodeNoFill`. And maybe even a log ?

Or you can reuse the `CR_CdbBid` above which contains the size. And to check if this is the expected kind of bid, maybe could you add a method `consumeFor:AdUnitType.Banner` like how it is done [here](https://codesearch.criteois.com/github.com/criteo/android-publisher-sdk/-/blob/publisher-sdk/src/main/java/com/criteo/publisher/Bid.java?L71)

Anyway, as you want to had such kind of logic, could you please add a test case with such scenario:
- Given a successful interstitial bid
- When displaying with a banner view
- Then nothing get displayed
- And listener get notified for failure

Reply:
Added the `consumeFor` method and unit tests.